storage/src/main/java/io/atomix/storage/buffer/FileBytes.java

```diff
@@ -412,6 +428,24 @@ public Bytes write(int position, byte[] bytes, int offset, int length) {
     return this;
   }
 
+  @Override
+  public Bytes write(int offset, ByteBuffer src, int srcOffset, int length) {
+    checkWrite(offset, length);
+
+    try {
+      final ByteBuffer view = src.asReadOnlyBuffer();
+      view.position(srcOffset).limit(srcOffset + length);
+      randomAccessFile.getChannel().write(view, offset);
+    } catch (IllegalArgumentException e) {
+      // this is necessary in order not to break the method contract
+      throw new IndexOutOfBoundsException(e.getMessage());
```

Comment:
:see_no_evil:

Reply:
:sweat_smile: it was either this or break the contract, I didn't have an other idea :see_no_evil: